ibis/backends/tests/test_signatures.py

```diff
@@ -0,0 +1,195 @@
+        ),
+    ),
+    "create_database": pytest.param(
+        CanCreateDatabase,
+        "create_database",
+        marks=pytest.mark.notyet(["clickhouse", "flink", "impala", "mysql", "pyspark"]),
+    ),
+    "drop_database": pytest.param(
+        CanCreateDatabase,
+        "drop_database",
+        marks=pytest.mark.notyet(["clickhouse", "impala", "mysql", "pyspark"]),
+    ),
+    "drop_table": pytest.param(
+        SQLBackend,
+        "drop_table",
+        marks=pytest.mark.notyet(
+            ["bigquery", "dask", "druid", "flink", "impala", "pandas", "polars"]
+        ),
+    ),
+    "execute": pytest.param(
+        SQLBackend,
+        "execute",
+        marks=pytest.mark.notyet(
+            ["clickhouse", "datafusion", "flink", "mysql", "pandas"]
+        ),
+    ),
+    "insert": pytest.param(
+        SQLBackend,
+        "insert",
+        marks=pytest.mark.notyet(["clickhouse", "flink", "impala", "sqlite"]),
+    ),
+    "list_databases": pytest.param(
+        CanCreateDatabase,
+        "list_databases",
+        marks=pytest.mark.notyet(
+            [
+                "clickhouse",
+                "flink",
+                "impala",
+                "mysql",
+                "postgres",
+                "risingwave",
+                "sqlite",
+            ]
+        ),
+    ),
+    "list_tables": pytest.param(
+        BaseBackend,
+        "list_tables",
+        marks=pytest.mark.notyet(
+            ["flink", "mysql", "oracle", "postgres", "risingwave"]
+        ),
+    ),
+    "read_csv": pytest.param(
+        BaseBackend,
+        "read_csv",
+        marks=pytest.mark.notyet(["dask", "duckdb", "flink", "pandas", "pyspark"]),
+    ),
+    "read_delta": pytest.param(
+        BaseBackend,
+        "read_delta",
+        marks=pytest.mark.notyet(["datafusion", "duckdb", "polars", "pyspark"]),
+    ),
+    "read_json": pytest.param(
+        BaseBackend,
+        "read_json",
+        marks=pytest.mark.notyet(["duckdb", "flink", "pyspark"]),
+    ),
+    "read_parquet": pytest.param(
+        BaseBackend,
+        "read_parquet",
+        marks=pytest.mark.notyet(["dask", "duckdb", "flink", "pandas"]),
+    ),
+    "table": pytest.param(
+        BaseBackend,
+        "table",
+        marks=pytest.mark.notyet(
+            [
+                "clickhouse",
+                "dask",
+                "datafusion",
+                "druid",
+                "duckdb",
+                "exasol",
+                "mssql",
+                "mysql",
+                "oracle",
+                "pandas",
+                "polars",
+                "postgres",
+                "risingwave",
+                "snowflake",
+                "sqlite",
+                "trino",
+                "pyspark",
+            ]
+        ),
+    ),
+    "to_parquet_dir": pytest.param(
+        BaseBackend,
+        "to_parquet_dir",
+        marks=pytest.mark.notyet(["pyspark"]),
+    ),
+    "truncate_table": pytest.param(
```

Comment:
So, basically the goal is to, over time, remove as many elements of this dict as possible?

Reply:
Ideally, we'd remove all of them.  There's a more generic "skip for all backends" list for things like `do_connect`, but I think we should be able to remove all of the entries on this list if we standardize positional keyword names, and enforce keyword-only arguments

Reply:
Also, once this is merged in, it will also prevent us from adding new methods with differing argument names (assuming those methods are defined in the base classes, which they should be)